### DISCUSSION TITLE
Loading UI Enhancements

### DISCUSSION THREAD

**Discussion by bigmistqke** [payloadcms/payload#1180] [Roadmap]
At the moment, when uploading a file, there is no visual feedback for how long the upload will take, while the screen is locked on the file-upload modal. This is not a pleasant UX.
![afbeelding](https://user-images.githubusercontent.com/10504064/192564510-00d42f8c-819a-43f5-b9be-11558f5d0836.png)

I propose to not lock the screen to the model, but to go straight back to the entry-page and make use of those bars at the bottom of the page (like 👇) to communicate the upload-progress.
![afbeelding](https://user-images.githubusercontent.com/10504064/192565003-6841775f-4fe3-4958-98f9-c42d26ba6d23.png)

If you try to leave the page before the file is uploaded, you could give a warning.
If you want to prevent bottleneck from uploading too many files at once, you could start queuing them if the amount goes over a certain threshold.

**Comment by qivox** [comment]
Generally having more loading indicators on the admin panel would be great. It sometimes feels a bit clunky changing routes or saving content for example.

**Comment by zubricks** [comment]
Hey @bigmistqke progress bar for uploads is 100% on our radar, and we'll be marking this feature as `planned` right now.

**Comment by rjdjohnston** [comment]
+1

**Comment by xndyz** [comment]
Any updates on this?

**Comment by teneburu** [comment]
Please 🙏

**Comment by JoshFinlayAU** [comment]
This is marked as roadmap:complete, however I cannot find any documentation for it?

**Comment by JoshFinlayAU** [comment]
https://github.com/payloadcms/payload/pull/1923 does not address roadmap 1180. It merely masks it. The roadmap item was for responsive upload progress (ie. percent, mb/kb uploaded, time remaining, etc).
